### PR TITLE
Add content on API policies

### PR DESCRIPTION
**Pull request by Nashaath**
## Purpose
Add content on API policies.

### CODE REVIEW

File: en/docs/api-management/api-policies.md [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+
+2. Deploying the API:
+
+     Once the deployment initiation is complete, you can specify configuration values if any, and proceed to deploy. Choreo builds the generated interceptor application and pushes the Docker image to the Docker registry. Finally, Choreo deploys the interceptor application with the mediation service and the API Proxy.
+
+When a mediation policy is attached to a specific flow, the API invocation undergoes the following behavioral modification:
+ 
+ ![Request/Response flow](../assets/img/api-management/api-policies/request-response-flow.png)
+
+ - In the request path, the requests that pass through the gateway reach the relevant component, and Choreo executes any attached policies to the resource's request path before sending it to the backend.
```
Comment (shaniR) [inline en/docs/api-management/api-policies.md:23]: Reaching the relevant component is quite confusing. shall we say Choreo instead?

Comment (shaniR) [reply en/docs/api-management/api-policies.md:23]: Also these policies are executed in the gateway right? I am not sure if we want to describe the components related to it?

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/docs/api-management/api-policies.md [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+You can apply a policy to alter the  `Request`, `Response`, or `Error` flow of an API invocation before it reaches the backend or the client. For example, you can add a policy to the response flow to transform the payload from JSON to XML and add a header to the response. 
+
+Choreo includes a set of default policies that address common use cases. You can easily attach one or more policies to an API proxy component implementation via the Choreo Console. If necessary, you can also rearrange or swap the policies you attach.
+
+In Choreo, when you attach a mediation policy to a proxy, the deployment is a two-step process.
+
+1. Deployment initiation: 
+
+     If the component to which you want to attach the mediation policy is new, the system creates and commits a new repository with the mediation service code based on the attached policy. This new service is called the interceptor application.
```
[the remark this answers is not in the document]
Comment (Nashaath) [reply en/docs/api-management/api-policies.md:13]: The user can see this in the deploy view so would be useful to understand what is happening I feel:
deploy

File: en/docs/api-management/api-policies.md [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+
+When a mediation policy is attached to a specific flow, the API invocation undergoes the following behavioral modification:
+ 
+ ![Request/Response flow](../assets/img/api-management/api-policies/request-response-flow.png)
+
+ - In the request path, the requests that pass through the gateway reach the relevant component, and Choreo executes any attached policies to the resource's request path before sending it to the backend. 
+
+- In the response path, the interceptor component receives response messages from the backend, and Choreo executes any mediation policies attached to the `Response` flow or the `Error` flow. Then the response is forwarded to the client.
+
+- If an error occurs during the execution of policies or due to an internal error, Choreo executes the `Error` flow and sends an error response to the client.
```
[the remark this answers is not in the document]
Comment (Nashaath) [reply en/docs/api-management/api-policies.md:27]: @menakaj can you plz confirm?

File: en/docs/api-management/api-policies.md [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+
+This policy validates the request and response headers with the configured values.
+
+```
+
+#### Policy versioning
+
+When it comes to policy versioning in Choreo or mediation dependencies, it is important to consider the major version changes in the Ballerina language. For example, when transitioning from update 1 to update 2 requires a major version increment, which can introduce significant incompatibilities.
+
+Therefore, to ensure compatibility, the recommended approach is to version the policy package in a manner that the major version gets upgraded when the Choreo/mediation dependency version is upgraded to a major version.
```
[the remark this answers is not in the document]
Comment (Nashaath) [reply en/docs/api-management/api-policies.md:383]: @menakaj can we add an example here to help the user understand the versioning aspect better?

**Review by shaniR — APPROVED** [review]

File: en/docs/api-management/api-policies.md (hunk 1 above)
Comment (menakaj) [reply en/docs/api-management/api-policies.md:23]: `requests that pass through the gateway reach the interceptor app, and executes any attached policies to the resource's request path before sending it to the backend.`

**Pull request merged**